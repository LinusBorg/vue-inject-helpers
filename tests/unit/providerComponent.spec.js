import { mount } from '@vue/test-utils'
import ProviderContainer from './resources/Provider'

describe('The Provider Component', () => {
  it('works', () => {
    const wrapper = mount(ProviderContainer)

    const span = wrapper.find('.test-span')
    expect(span.text()).toBe('test message')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('re-renders when reactive data is changed', () => {
    const wrapper = mount(ProviderContainer)

    wrapper.setData({
      msg: 'new test message',
    })

    const span = wrapper.find('.test-span')
    expect(span.text()).toBe('new test message')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
