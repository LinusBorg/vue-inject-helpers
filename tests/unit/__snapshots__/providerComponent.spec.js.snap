// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Provider Component re-renders when reactive data is changed 1`] = `
<div>
  <div><span class="test-span">new test message</span></div>
</div>
`;

exports[`The Provider Component works 1`] = `
<div>
  <div><span class="test-span">test message</span></div>
</div>
`;
